app/forms/state_file/ny_sales_use_tax_form.rb

```diff
@@ -0,0 +1,32 @@
+module StateFile
+  class NySalesUseTaxForm < QuestionsForm
+    set_attributes_for :intake,
+                       :untaxed_out_of_state_purchases,
+                       :sales_use_tax_calculation_method,
+                       :sales_use_tax
+
+    validates :untaxed_out_of_state_purchases, presence: true
+    validates :sales_use_tax_calculation_method, presence: true, if: -> { untaxed_out_of_state_purchases == "yes" }
+    validates :sales_use_tax, presence: true, if: -> { sales_use_tax_calculation_method == "manual" }
+
+    def initialize(intake = nil, params = nil)
+      if params[:untaxed_out_of_state_purchases] == "no"
+        params[:sales_use_tax_calculation_method] = "unfilled"
+        params[:sales_use_tax] = nil
+      end
+      if params[:sales_use_tax_calculation_method] == "automated"
+        params[:sales_use_tax] = intake.calculate_sales_use_tax
+      end
+      super(intake, params)
+    end
```

Comment:
I think the before_save hook on the model will prevent bad data from going, but I appreciate that this will update the form to reflect the user's choices if they have a validation error and it rerenders the page. Very thoughtful of you to include this.